go/pkg/tool/tool.go

```diff
@@ -31,6 +36,102 @@ type Client struct {
 	GrpcClient *rc.Client
 }
 
+// ReexecuteAction reexecutes the action remotely, optionally overriding the
+// inputs with the ones provided at the inputRoot path.
+func (c *Client) ReexecuteAction(ctx context.Context, actionDigest, inputRoot string, oe outerr.OutErr) error {
+	acDg, err := digest.NewFromString(actionDigest)
+	if err != nil {
+		return err
+	}
+	actionProto := &repb.Action{}
+	if err := c.GrpcClient.ReadProto(ctx, acDg, actionProto); err != nil {
+		return err
+	}
+
+	commandProto := &repb.Command{}
+	cmdDg, err := digest.NewFromProto(actionProto.GetCommandDigest())
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Reading command from action digest..")
+	if err := c.GrpcClient.ReadProto(ctx, cmdDg, commandProto); err != nil {
+		return err
+	}
+
+	fmc := filemetadata.NewNoopCache()
+	client := &rexec.Client{
+		FileMetadataCache: fmc,
+		GrpcClient:        c.GrpcClient,
+	}
+	_, inputPaths, err := c.getInputTree(ctx, actionProto.GetInputRootDigest())
+	if err != nil {
+		return err
+	}
+	if inputRoot == "" {
+		curTime := time.Now().Format(time.RFC3339)
+		inputRoot = filepath.Join(os.TempDir(), acDg.Hash+"_"+curTime)
```

Comment:
Why not `ioutil.TempDir(os.TempDir(), acDg.Hash + "_" + curTime)` ?

Reply:
Only because it's one line longer (ioutil.TempDir returns an error), and I don't need to worry about races here (since it's a manual tool).